pkg/api/handlers/threads.go

```diff
@@ -230,7 +230,9 @@ func (a *ThreadHandler) Update(req api.Context) error {
 			return err
 		}
 		for _, newTool := range newThread.Tools {
-			if !slices.Contains(agent.Spec.Manifest.AvailableThreadTools, newTool) {
+			possibleTools := slices.Concat(agent.Spec.Manifest.AvailableThreadTools, agent.Spec.Manifest.DefaultThreadTools)
+
+			if !slices.Contains(possibleTools, newTool) {
```

Comment:
This is fine. A small change would save allocating a whole new slice here.
```suggestion
			possibleTools := slices.Concat(agent.Spec.Manifest.AvailableThreadTools, agent.Spec.Manifest.DefaultThreadTools)

			if !slices.Contains(agent.Spec.Manifest.AvailableThreadTools, newTool) && !slices.Contains(agent.Spec.Manifest.DefaultThreadTools, newTool) {
```

Reply:
done